src/main/scala/com/microsoft/kusto/spark/utils/DeviceAuthentication.scala

```diff
@@ -0,0 +1,67 @@
+package com.microsoft.kusto.spark.utils
+
+import java.awt.Desktop
+import java.net.URI
+import java.util.concurrent.{ExecutionException, ExecutorService, Executors, Future}
+
+import com.microsoft.aad.adal4j.{AuthenticationContext, AuthenticationResult, DeviceCode}
+import javax.naming.ServiceUnavailableException
+import org.apache.log4j.{Level, Logger}
+import java.awt.Toolkit
+import java.awt.datatransfer.StringSelection
+import java.awt.datatransfer.DataFlavor
+
+
+object DeviceAuthentication {
+  private val CLIENT_ID = "db662dc1-0cfe-4e1c-a843-19a68e65be58"
```

Comment:
What is it? Is it guaranteed to be preserved?

Reply:
yes